internal/build/phase_config_provider.go

```diff
@@ -108,11 +111,33 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
+
 		bind := "/var/run/docker.sock:/var/run/docker.sock"
+
+		if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+			if strings.HasPrefix(dockerHost, "unix://") {
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			}
+			if strings.HasPrefix(dockerHost, "tcp://") {
+				bind = ""
+				if runtime.GOOS == "linux" {
+					provider.hostConf.NetworkMode = "host"
```

Comment:
So, to clarify, this is specifically for SELinux? (https://github.com/buildpacks/pack/pull/988#issuecomment-744619639)

Reply:
My primary motivation to support `TCP` is `SELinux`. I found it easier to expose `docker API` to container that way. Of course user can set proper `SELinux` policy to make it work with `Unix` socket (some docker installation pkgs do that), but I think that's little bit less convenient. Beside `TCP` may be useful if user want's to run build on remote machine.

Reply:
The `runtime.GOOS` would need to be `provider.os` instead, though I think the check could be omitted entirely (and the implementation would otherwise work on Windows) without the `provider.hostConf.NetworkMode=`.

Additionally, I feel like we may want to omit the `provider.hostConf.NetworkMode=`, since `pack build` already has a `--network` flag, and any user setting `DOCKER_HOST=tcp://` will need to consider remote host/container concerns like volumes, bind mounts anyways, in addition to networking.

Also, I wonder if we lose a use-case by always setting the mode as `host`: a remote docker host whose containers should have no network access.